### PR TITLE
Types#21

### PR DESCRIPTION
**Pull request by sculove**
### issues
#21 

### task
- provides `component.d.ts` (typescirpt definition)
- update dependency files

### CODE REVIEW

File: src/component.d.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace eg;
+
+export = Component;
+declare class Component {
+  constructor(options?: { [key: string]: any });
+  trigger(eventName: string, customEvent?: { [key: string]: any }): boolean;
+  hasOn(eventName: string): boolean;
+  on(eventName: string, handlerToAttach: (event?: { [key: string]: any }) => any): Component;
```
Comment (mixed) [inline src/component.d.ts]: A Event always come in as parameters. So.. All `events` should be changed to` event`.
`event?` -> `event`

File: src/component.d.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  on(events: { [key: string]: (event?: { [key: string]: any }) => any }): Component;
+  off(eventName: string, handlerToAttach: (event?: { [key: string]: any }) => any): Component;
+  off(eventName: string): Component;
+  off(): Component;
+  once(events: { [key: string]: (event?: { [key: string]: any }) => any }): Component;
+  once(eventName: string, handlerToAttach: (event?: { [key: string]: any }) => any): Component;
```
Comment (mixed) [inline src/component.d.ts]: (nit)
Unlike on and off, the `define` position of the grouping method has changed.

File: src/component.d.ts (hunk 1 above)
Comment (sculove) [inline src/component.d.ts]: @mixed ok. thanks

**Comment by sculove** [comment]
I added `options` property.

File: src/component.d.ts (hunk 2 above)
Comment (sculove) [inline src/component.d.ts]: @mixed I don't understand. Could you give me an example?

**Pull request merged**